From 2bf76dc97e65611bcbc5bb82a264ae6a88df8903 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Thu, 2 Nov 2023 10:39:34 -0400
Subject: [PATCH 52/68] v4l2codecs: encoder: Save the sink format

Save the V4L2 sink format that has been selected. The information
enclosed will be used to validate buffer importation.
Upstream-Status: Pending
---
 sys/v4l2codecs/gstv4l2encoder.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2codecs/gstv4l2encoder.c b/sys/v4l2codecs/gstv4l2encoder.c
index 0886d15..a9e0939 100644
--- a/sys/v4l2codecs/gstv4l2encoder.c
+++ b/sys/v4l2codecs/gstv4l2encoder.c
@@ -77,8 +77,9 @@ struct _GstV4l2Encoder
   GstQueueArray *pending_requests;
   guint version;
 
-  enum v4l2_buf_type src_buf_type;
   enum v4l2_buf_type sink_buf_type;
+  struct v4l2_format sink_fmt;
+  enum v4l2_buf_type src_buf_type;
   gboolean mplane;
 
   /* properties */
@@ -575,6 +576,7 @@ gst_v4l2_encoder_select_sink_format (GstV4l2Encoder * self, GstVideoInfo * in,
 
   GST_INFO_OBJECT (self, "Selected sink format %s %ix%i",
       gst_video_format_to_string (out->finfo->format), out->width, out->height);
+  self->sink_fmt = fmt;
 
   return TRUE;
 }
-- 
2.25.1

